src/engine.rs

```diff
@@ -523,9 +543,13 @@ impl ApplyBackend for Engine {
                     reset_storage,
                 } => {
                     Engine::set_nonce(&address, &basic.nonce);
+
+                    //TODO: check this part
                     // Apply changes for eth-connector
                     EthConnectorContract::get_instance()
                         .internal_set_eth_balance(&address, &basic.balance);
+                    Engine::set_balance(&address, &Wei::new(basic.balance));
```

Comment:
Looks like we need to refactor `internal_set_eth_balance` to take the `Wei` newtype. And we can remove the direct `set_balance` call.